### PR TITLE
Multi-cluster configuration example w/ kustomize.

### PR DESCRIPTION
**Pull request by abutcher**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+        - [Builds](#Builds)
+        - [Registries](#Registries)
+        - [Console](#Console)
+        - [Scheduler Policy](#Scheduler-Policy)
+        - [Machine Sets](#Machine-Sets)
+- [Multi-cluster Management](#Multi-cluster-Management)
+    - [Deploy Configuration to Multiple Clusters](#Deploy-Configuration-to-Multiple-Clusters)
+    - [Customizing Configuration By Cluster](#Customizing-Configuration-By-Cluster)
+
+<!-- /TOC -->
```
Comment (dgoodwin) [inline README.md:21]: What are you using to generate these?

File: README.md [hunk 2]
```diff
@@ -123,3 +143,227 @@ A standard OpenShift 4 cluster with 3 compute nodes in us-east-1 comes with 6 Ma
 
 TODO: Should we recommend against using MachineSets with gitops and Argo? Or is there a templating solution we should explore? In this case the value we want to template is a fact about the individual cluster it's being deployed to.
 
+# Multi-cluster Management
+
+In this example we will manage the build configuration of two OpenShift 4.x clusters, a pre-production (context: `pre`) cluster and a production (context: `pro`) cluster.
+
+The example build configuration we will deploy is in a separate git repository [https://github.com/abutcher/gitops-builds-example](https://github.com/abutcher/gitops-builds-example) that contains customizations to be made per cluster environment.
```
Comment (dgoodwin) [inline README.md]: Can you elaborate on the structure of that repo, how kustomize is being used and the branching strategy? I am confused why different branches are needed, I would have expected a single branch, with a base and an overlay structure.

File: README.md [hunk 3]
```diff
@@ -123,3 +143,227 @@ A standard OpenShift 4 cluster with 3 compute nodes in us-east-1 comes with 6 Ma
...
+                   --dest-namespace=openshift-config \
+                   --revision pre
+
+$ argocd app create --project default \
+                   --name pro-builds \
+                   --repo https://github.com/abutcher/gitops-builds-example.git \
+                   --path base \
+                   --dest-server https://api.cluster2.new-installer.openshift.com:6443 \
+                   --dest-namespace=openshift-config \
+                   --revision pro
```
Comment (dgoodwin) [inline README.md]: Per above I'm not sure why different branches are involved, I thought the kustomize dirs did that for us.

File: README.md [hunk 4]
```diff
@@ -123,3 +143,227 @@ A standard OpenShift 4 cluster with 3 compute nodes in us-east-1 comes with 6 Ma
...
 
+# Multi-cluster Management
+
+In this example we will manage the build configuration of two OpenShift 4.x clusters, a pre-production (context: `pre`) cluster and a production (context: `pro`) cluster.
+
+The example build configuration we will deploy is in a separate git repository [https://github.com/abutcher/gitops-builds-example](https://github.com/abutcher/gitops-builds-example) that contains customizations to be made per cluster environment.
+
+## Deploy Configuration to Multiple Clusters
+
+Ensure we have access to both clusters and that ArgoCD has been deployed to each,
```
Comment (dgoodwin) [inline README.md]: More common usecase would probably be one argocd with multiple clusters registered, but I'm not sure it's important enough to change.

File: README.md (hunk 1 above)
Comment (abutcher) [reply README.md:21]: I did it by hand but https://github.com/ekalinin/github-markdown-toc generates the same thing if we want to generate. I could add a hack/makefile situation.

File: README.md (hunk 2 above)
Comment (abutcher) [reply README.md]: Removed branches in latest update w/ a directory hierarchy.

File: README.md (hunk 4 above)
Comment (abutcher) [reply README.md]: You're right. ArgoCD on my hive cluster w/ configured clusters works fine.

File: README.md [hunk 5]
```diff
@@ -123,3 +143,211 @@ A standard OpenShift 4 cluster with 3 compute nodes in us-east-1 comes with 6 Ma
...
+Name:               pro-kustomize-builds
+Project:            default
+Server:             https://api.cluster2.new-installer.openshift.com:6443
+Namespace:          openshift-config
+URL:                https://argocd-server-argocd.apps.cluster2.new-installer.openshift.com/applications/pro-kustomize-builds
+Repo:               https://github.com/dgoodwin/openshift4-gitops.git
+Target:             pro
+Path:               builds/overlays/pro
+Sync Policy:        Automated
+Sync Status:        Synced to pro (9a11bf9)
```
Comment (dgoodwin) [inline README.md]: Should this hash be the same as above now that we're on the same branch?

File: README.md [hunk 6]
```diff
@@ -123,3 +143,211 @@ A standard OpenShift 4 cluster with 3 compute nodes in us-east-1 comes with 6 Ma
...
+https://api.cluster1.new-installer.openshift.com:6443   pre   Successful
+```
+
+Add our build configuration repository to ArgoCD. The build configuration repository has a `pre` and `pro` kustomize overlay which will override the build `imageLabels` by cluster but we will start by deploying the base build configuration.
+
+```bash
+$ argocd repo add https://github.com/dgoodwin/openshift4-gitops.git
+```
+
+Deploy custom OpenShift build configuration to pre-production and production clusters,
```
Comment (dgoodwin) [inline README.md:235]: it feels like this section is redundant with the example below in customizing by cluster. Was this leftover or is there some value in this part I'm not seeing. It seems like it's just applying the same thing to two clusters.

**Review by dgoodwin — COMMENTED** [review]
This is making more sense to me now, looks good.

File: README.md (hunk 6 above)
Comment (abutcher) [reply README.md:235]: Yeah, it’s just the same thing to two clusters. I can cut it if you don’t feel like it adds anything.

File: README.md (hunk 5 above)
Comment (abutcher) [reply README.md]: I’ll update this.

**Pull request merged**